File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagnamaster

// TODO: define the 'PreparationTime()' function
func PreparationTime(layers []string, timePerLayer int) int {
    if timePerLayer == 0 {
        timePerLayer = 2 
    }

    estimatedTime := len(layers) * timePerLayer
    return estimatedTime
}

// TODO: define the 'Quantities()' function
func Quantities(layers []string) (int, float64) {
    noodles := 0 
    sauces := 0.0 

    for _, layer := range layers {
        switch layer {
            case "noodles": 
            	noodles += 50
            case "sauce": 
            	sauces += 0.2
        }
    }

    return noodles, sauces
}

// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(friendList []string, myList []string){
    myList[len(myList)-1] = friendList[len(friendList)-1]
}

// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(amounts []float64, portions int) []float64 {

    factor := float64(portions) / 2

    results := make([]float64, len(amounts))

    for i, amount := range amounts {
        results[i] = amount * factor
    }

    
    return results
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
//
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more
// functionality.

File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */

export function cookingStatus(timerInMinutes) {
  if (timerInMinutes === undefined) {
    return 'You forgot to set the timer.'
  } 

  if (timerInMinutes === 0) {
    return "Lasagna is done."
  }

  return "Not done, please wait."
}

export function preparationTime(layers, numOfLayers = 2) {
  return layers.length * numOfLayers
}

export function quantities(layers) {
  let noodles = 0;
  let sauce = 0;

  for (let i = 0; i < layers.length; i++) {
    switch (layers[i]) {
      case 'noodles':
        noodles += 50;
        break;
      case 'sauce':
        sauce += 0.2;
        break;
    }
  }

  return {
    noodles,
    sauce,
  };
}

export function addSecretIngredient(friendList, myList) {
  const secretIngredient = friendList[friendList.length - 1];
  myList.push(secretIngredient);
}

export function scaleRecipe(recipe, numOfScaling) {

  var scaled = {}
  const portion = numOfScaling / 2
  
  for (const ingredient in recipe) {
    scaled[ingredient] = recipe[ingredient] * portion
  }

  return scaled
}
